### ISSUE TITLE
Assign Recruit List - Gone | Annoying text in Scenario Editor

### ISSUE THREAD

**Issue by FrostBlade5** [wesnoth/wesnoth#7020]
### Describe the desired feature

![assign recruit list gone](https://user-images.githubusercontent.com/91756004/192092267-55a9aa35-eb1a-4fee-8f21-59d64af308d6.png)

Whoever had the brilliant idea to remove assign recruit list feature, why just why??? Either i am to dumb to figure out where it went or it has been removed, why would you?!

![remove txt in map editor](https://user-images.githubusercontent.com/91756004/192092296-b7087aea-b42f-420f-a08d-716c43d22f98.png)

Can this annoying text at the bottom of the screen be removed somehow? It is blocking my sight on the map which is bad for editing, besides i don't need to read this 24/7

**Comment by Wedge009** [comment]
If you desire people to pay attention to your request, I encourage you to rewrite it in a clearer and more respectful fashion.

**Comment by Descacharrado** [comment]
The assign recruit list was removed in #5364 but as https://github.com/wesnoth/wesnoth/pull/5364#issuecomment-746412146 said, it would be a good feature with a proper UI.
The FR about the editor text is a duplicate of #6422

**Comment by FrostBlade5** [comment]
> If you desire people to pay attention to your request, I encourage you to rewrite it in a clearer and more respectful fashion.

I don't know what you mean by clearer, i think both issues i mentioned are easy to understand.
Also i did not inted to sound disrespectful, i just get to the point and say what i think even if some people don't like that.
See, the assign recruit list option was a very useful feature and there is no reason to remove that. You can say this is just my opinion but the fact of the matter is, there is no reason to remove a useful option.

If i remember correctly it was removed because people kept asking how to use it. Well...

**Comment by FrostBlade5** [comment]
![Assign Recruit List](https://user-images.githubusercontent.com/91756004/192123591-fd945a97-70e2-4a01-bf68-7a3360e03bfd.PNG)
![7](https://user-images.githubusercontent.com/91756004/192123594-a2936da3-b20b-426e-922c-d082216cfc86.png)
![1](https://user-images.githubusercontent.com/91756004/192123598-6932d878-d45a-43d1-adf2-3eed0bb42cb0.png)
![2](https://user-images.githubusercontent.com/91756004/192123599-02d7e810-d1f0-4ec5-8eb7-1232ce17e9fa.png)
![3](https://user-images.githubusercontent.com/91756004/192123601-b9b71fd0-581e-48cd-a9f5-75f05277f1f2.png)
![4](https://user-images.githubusercontent.com/91756004/192123602-1268310f-2b4d-436c-a427-0f588e65b064.png)
![5](https://user-images.githubusercontent.com/91756004/192123604-fdc88bcd-2683-4465-8ddd-c5623bf2e002.png)
![6](https://user-images.githubusercontent.com/91756004/192123608-f6475eb4-9002-4af5-85a1-29849df9de29.png)

**Comment by CelticMinstrel** [comment]
> See, the assign recruit list option was a very useful feature and there is no reason to remove that.

It was removed because:

1. The way it worked was extremely confusing and buggy and
2. The person removing it didn't have the time and/or understanding to fix it.

There is no opposition to adding the feature back at a later time as long as it can be done in a way that makes sense and doesn't give bizarre answers if you make a mistake. The instructions you wrote on how to use it indicate that the feature was not functioning how it should.

* There should be no need to place a unit to assign a recruit list. Recruit list as a feature of the side, rather than a specific unit. (Specific units _can_ have their own recruit lists to supplement the side's recruit list, but that only makes sense if you have multiple leaders.)
* The need to explicitly remove the Ancient Lich is a major problem. You should not have to take specific action to make the game not do something that almost no-one would want the game to do.
* The way you describe assigning the list doesn't make sense. You select units, _then_ assign them? The proper way to handle this would be for Assign Recruit List to bring up a dialog where you choose which units go on the list, or something more along those lines.
* Needing to add `faction=Custom` is a obvious issue as well.

In short, there were numerous bugs with the system, and someone decided it would be easier to delete it entirely and remake it from scratch rather than fixing the bugs.

**Comment by Pentarctagon** [comment]
> There should be no need to place a unit to assign a recruit list. Recruit list as a feature of the side, rather than a specific unit. (Specific units can have their own recruit lists to supplement the side's recruit list, but that only makes sense if you have multiple leaders.)

This was nevertheless what the editor was doing - it didn't assign a recruit list to a side, it assigned a recruit list to a specific unit via `extra_recruit`.

**Comment by FrostBlade5** [comment]
> > This was nevertheless what the editor was doing - it didn't assign a recruit list to a side, it assigned a recruit list to a specific unit via `extra_recruit`.

Exactly, i know that each side/race has their specific recruit list, i wanted to add a recruit list of individual selected units to a specific unit. For those people that are aware about faction=Custom and the ancient-lich, the ingame option was the easier way, as now we have to edit the cfg file, which is not that much work but it is extra work.

What i would have done with the assign recruit list option if i had the knownledge, remove the preselected anicent lich and make it so that whenever the creator presses assign recruit list this will automatically add faction=Custom to the cfg file.
Problem solved? I admit tho that insturctions how how to use the custom reruit list could be better guided with a few text hints like a one time tutorial that can be triggered manually if needed.

My issue is now, as the option was removed i can't add units and read their exact name/id in the file afterwards, now i have to manually add all units to the custom recruit list but how do i do that when i am unaware of the exact units names/id i want to add? Guess there is a list of precise unitnames sowhere out there?

Again i think the ancientlich and faction=custom are the only problems really, or recreate it from scratch, anyways i see this will take some time before anyone is even going to touch this again...
